src/test/java/org/isf/medicalsinventory/test/Tests.java

```diff
@@ -0,0 +1,220 @@
+		testMedicalInventory = new TestMedicalInventory();
+		testWard = new TestWard();
+		testMedicalInventoryRow = new TestMedicalInventoryRow();
+		testMedical = new TestMedical();
+		testLot = new TestLot();
+		testMedicalType = new TestMedicalType();
+	}
+	
+	@Before
+	public void setUp() {
+		cleanH2InMemoryDb();
+	}
+	
+	@Test
+	public void testMedicalInventoryGets() throws Exception {
+		int code = setupTestMedicalInventory(false);
+		checkMedicalInventoryIntoDb(code);
+	}
+	
+	@Test
+	public void testIoGetMedicalInventorys() throws Exception {
+		int id = setupTestMedicalInventory(false);
+		MedicalInventory foundMedicalinventory = medIvnIoOperationRepository.findById(id).orElse(null);
+		assertThat(foundMedicalinventory).isNotNull();
+		List<MedicalInventory> medicalInventories = medicalInventoryIoOperation.getMedicalInventory();
+		assertThat(medicalInventories.get(medicalInventories.size() - 1).getId()).isEqualTo((Integer) id);
+	}
+	
+	@Test
+	public void testIoNewMedicalInventory() throws Exception {
+		Ward ward = testWard.setup(false);
+		MedicalInventory medicalInventory = testMedicalInventory.setup(ward, false);
+		MedicalInventory newMedicalInventory = medicalInventoryIoOperation.newMedicalInventory(medicalInventory);
+		checkMedicalInventoryIntoDb(newMedicalInventory.getId());
+	}
+	
+	@Test
+	public void testIoUpdateMedicalInventory() throws Exception {
+		Integer id = setupTestMedicalInventory(false);
+		MedicalInventory foundMedicalInventory = medIvnIoOperationRepository.findById(id).orElse(null);
+		assertThat(foundMedicalInventory).isNotNull();
+		String status = "canceled";
+		foundMedicalInventory.setStatus(status);
+		MedicalInventory updatedMedicalInventory = medicalInventoryIoOperation.updateMedicalInventoryRow(foundMedicalInventory);
+		assertThat(updatedMedicalInventory.getStatus()).isEqualTo(status);
+	}
+	
+	@Test
+	public void testIoDeleteMedicalInventory() throws Exception {
+		Integer id = setupTestMedicalInventory(false);
+		MedicalInventory foundMedicalInventory = medIvnIoOperationRepository.findById(id).orElse(null);
+		assertThat(foundMedicalInventory).isNotNull();
+		String reference = foundMedicalInventory.getInventoryReference();
+		medicalInventoryIoOperation.deleteMedicalInventory(foundMedicalInventory);
+		assertThat(medicalInventoryIoOperation.referenceExists(reference)).isFalse();
+	}
+	
+	@Test
+	public void testIoGetMedicalInventoryWithStatus() throws Exception {
+		int id = setupTestMedicalInventory(false);
+		MedicalInventory foundMedicalInventory = medIvnIoOperationRepository.findById(id).orElse(null);
+		assertThat(foundMedicalInventory).isNotNull();
+		List<MedicalInventory> medicalinventories = medicalInventoryIoOperation
+				.getMedicalInventoryByStatus(foundMedicalInventory.getStatus());
+		assertThat(medicalinventories.get(0).getStatus()).isEqualTo(foundMedicalInventory.getStatus());
+	}
+	
+	@Test
+	public void testIoGetMedicalInventoryWithStatusAndWard() throws Exception {
```

Comment:
Here I would test the filter by status, so I would saveAndFlush 2 or 3 inventories with different ward and statuses and then check the result

Reply:
If you need you can also save only one and then retrieve it by status and ward and then check if the retrieve result has the right status and ward

Reply:
okay I can also add another inventory before retrieve inventory. okay let me add another inventory

Reply:
yes I mean check the result of the method under testing (not of the save and flush operation itself)

Reply:
I would also check that the result size is 1 because the `secondMedicalInventory` should have been filtered out.
If the size is 2, the first element could be still `firstMedicalInventory` but the method would not filtering anything...

Reply:
okay